src/States/MapViewStateUi.cpp

```diff
@@ -208,15 +208,15 @@ void MapViewState::setupUiPositions(int w, int h)
 	mStructures.iconMargin(constants::MARGIN_TIGHT);
 
 	// Anchored window positions
-	mFileIoDialog.position(centerWindowWidth(mFileIoDialog.width()), 50);
-	mGameOverDialog.position(centerWindowWidth(mGameOverDialog.width()), centerWindowHeight(mGameOverDialog.height()) - 100);
-	mAnnouncement.position(centerWindowWidth(mAnnouncement.width()), centerWindowHeight(mAnnouncement.height()) - 100);
-	mGameOptionsDialog.position(centerWindowWidth(mGameOptionsDialog.width()), centerWindowHeight(mGameOptionsDialog.height()) - 100);
+	mFileIoDialog.position(centerWindowWidth(mFileIoDialog.width()), 50.0f);
+	mGameOverDialog.position(centerWindowWidth(mGameOverDialog.width()), centerWindowHeight(mGameOverDialog.height()) - 100.0f);
+	mAnnouncement.position(centerWindowWidth(mAnnouncement.width()), centerWindowHeight(mAnnouncement.height()) - 100.0f);
+	mGameOptionsDialog.position(centerWindowWidth(mGameOptionsDialog.width()), centerWindowHeight(mGameOptionsDialog.height()) - 100.0f);
 
-	mDiggerDirection.position(centerWindowWidth(mDiggerDirection.width()), static_cast<int>(h / 2) - 125);
+	mDiggerDirection.position(centerWindowWidth(mDiggerDirection.width()), static_cast<int>(h / 2.0f) - 125.0f);
```

Comment:
And again.

To be clear, the casting is done whenever a division is used to determine placement of a UI element and to force pixel alignment. I suppose this could be moved up to Control to not make it the users problem but that would be a different change set.

I'm open to discussion on this -- maybe it would make sense to open an issue and have the pixel alignment handled directly in Control's position assignment operations.

Reply:
Maybe we should be using [`std::floor`](https://en.cppreference.com/w/cpp/numeric/math/floor) if what we really want is rounding. I assume in this case the type is unchanged, so both pre-rounded and post-rounded values are `float`.

More long term, I think we should be using integer coordinates for UI component positions, particularly if the values basically represent pixels anyway.

Reply:
In my experience, using percentages and unit offsets for UI alignment performs better on a wider variety of screen resolutions. So say you want a control to appear in the center of its parent, then offset by a few units down. The call would be: `setPosition(0.5f, 0.5f, 0.0f, 2.0f)`. As a bonus, this is resolution independent. It will look the same for any screen.

This is most definitely long term though as it involves setting up a matrix transform hierarchy and we don't have support for matrices yet.